### DISCUSSION TITLE
Turn 3D point cloud into smooth surface, currently having issues

### DISCUSSION THREAD

**Discussion by garciachristian829** [pyvista/pyvista#1959] [Q&A] [answered]
I am trying to turn a 3D point cloud into a surface but running into the following issues:

Code: 
cloud = pv.PolyData(xyz)
surf = cloud.delaunay_2d()
surf.plot(show_edges=True)

 - The 1/4 circle as shown in the image should not be filled. How can I avoid this?
- My mesh is very rough, I want to smooth it but not sure how. Is this even possible?
- I've tried reconstruct_surface but that does not work, instead it drapes the points in a 2D space. 

I would greatly appreciate anyone's help with these issues.

![PyVista Get Rid of Hole Seal](https://user-images.githubusercontent.com/31256654/146965931-62a07850-fd5d-4c55-a3a6-dd91bc022bfe.JPG)

**Comment by RichardScottOZ** [comment]
What does it look like as a cloud?

**Reply by garciachristian829** [reply]
This problem was actually solved on stackoverflow, here is the answer: https://stackoverflow.com/questions/70492211/3d-surface-in-pyvista-is-not-generating-correctly-i-am-trying-to-avoid-closing/70530662?noredirect=1#comment124714462_70530662

**Comment by garciachristian829** [comment, marked as answer]
https://stackoverflow.com/questions/70492211/3d-surface-in-pyvista-is-not-generating-correctly-i-am-trying-to-avoid-closing/70530662?noredirect=1#comment124714462_70530662
